spec/Section 5 -- Validation.md

```diff
@@ -1304,15 +1304,26 @@ fragment resourceFragment on Resource {
 
 **Formal Specification**
 
-- For each input Value {value} in the document:
+- For each literal Input Value {value} in the document:
   - Let {type} be the type expected in the position {value} is found.
-  - {value} must be coercible to {type}.
+  - {value} must be coercible to {type} (with the assumption that any
+    {variableUsage} nested within {value} will represent a runtime value valid
+    for usage in its position).
```

Comment:
What about custom scalars? Should they also be skipped as their coercing is implementation dependant? 

It's not an issue for a server but GraphiQL for an example cannot really enforce this without knowledge about the implementation that is not available in introspection.

Reply:
To GraphiQL, all custom scalar values should be "valid" since it can include them in the request it sends to the server. The server should validate them.

Reply:
GraphiQL cannot really say whether `"InvalidDate"` is coercible to a `Date` in literal positions.

For an example, with this operation:

```
{
  events(after: "InvalidDate") {
    title
    startsAt
  }
}
```

If we read the spec strictly, this should trigger a validation error. If GraphiQL doesn't highlight it, then there's a discrepency. 

From an Apollo Kotlin point of view, this is also an issue because validating those literal positions would require the users to pass the coercing to the compiler, which is extra complexity. Doable, for sure, but not trivial.

Reply:
Follow up PR: https://github.com/graphql/graphql-spec/pull/1157